test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,221 @@
+                if entries[2] != "cgroup":
+                    continue
+                flags = entries[3].split(',')
+                subsys = os.path.basename(entries[1])
+                paths[subsys] = entries[1]
+                if 'memory' in flags:
+                    paths['memsw'] = paths['memory']
+        return paths
+
+    def load_hook(self, filename):
+        """
+        Import and enable a hook pointed to by the URL specified.
+        """
+        try:
+            with open(filename, 'r') as fd:
+                script = fd.read()
+        except IOError:
+            self.assertTrue(False, "Failed to open hook file %s" % filename)
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
+
+    def load_config(self, cfg):
+        """
+        Create a hook configuration file with the provided contents.
+        """
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=5,
+                           starttime=self.server.ctime)
+        self.logger.info("Current config: %s" % cfg)
+        # Restart MoM to work around PP-993
+        self.mom.restart()
+
+    @timeout(1800)
+    def test_cgroups_epilogue_race(self):
+        """
+        Test to verify cgroups hook does not read/write  needed files
+        """
+        pcpus = 0
+        with open('/proc/cpuinfo', 'r') as desc:
+            for line in desc:
+                if re.match('^processor', line):
+                    pcpus += 1
+        if pcpus < 8:
+            self.skipTest("Test requires at least 8 physical CPUs")
+
+        attr = {'job_history_enable': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+        self.load_config(self.cfg0)
+        now = time.time()
+        j = Job(TEST_USER, attrs={ATTR_J: '1-5000'})
+        j.create_script(self.true_script)
+        jid = self.server.submit(j)
+        jid = jid.split('[')
+        for i in range(1, 1001):
+            if i % 100 == 0:
+                self.mom.log_match(msg="IOError", starttime=now,
+                                   existence=False, max_attempts=1, n="ALL")
+                now = time.time()
+            sjid = jid[0] + "[" + str(i) + jid[1]
```

Comment:
This is an interesting way of creating subjobs. I usually use following
jid.replace("[","[%s" %str(i))

Reply:
I think mine is faster, since I only search the string once outside of the loop, then build it for each subjob.